app/lib/efile/nj/nj1040_calculator.rb

```diff
@@ -302,10 +306,28 @@ def calculate_line_43
         should_use_property_tax_deduction ? calculate_tax_liability_with_deduction.round : calculate_tax_liability_without_deduction.round
       end
 
+      def calculate_line_45
+        calculate_line_43
+      end
+
+      def calculate_line_49
+        0
+      end
+
+      def calculate_line_50
+        difference = calculate_line_45 - calculate_line_49
+        difference.positive? ? difference : 0
+      end
+
       def calculate_line_51
         (@intake.sales_use_tax || 0).round
       end
 
+      def calculate_line_54
+        sum = calculate_line_50 + calculate_line_51
+        sum.positive? ? sum : 0
```

Comment:
[dust] truly no preference from my end, and i'm not the most aware of the ruby idioms, but `[sum, 0].max` would be equivalent if you want